<!doctype html>
<html class="no-js" lang="en">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />

		<title>bonohu blog</title>
		<meta name="description" content="">
		<meta name="author" content="bonohu">

		<link rel="stylesheet" href="/theme/css/foundation.css" />
		<link rel="stylesheet" href="/theme/css/pygment/monokai.css" />
		<link rel="stylesheet" href="/theme/css/custom.css" />


		<script src="/theme/js/modernizr.js"></script>

		<!-- Feeds -->

<script async src="https://www.googletagmanager.com/gtag/js?id=UA-113297-11"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-113297-11');
</script>

		<!-- mathjax config similar to math.stackexchange -->
		<script type="text/x-mathjax-config">
		MathJax.Hub.Config({
			jax: ["input/TeX", "output/HTML-CSS"],
			tex2jax: {
				inlineMath: [ ['$', '$'] ],
				displayMath: [ ['$$', '$$']],
				processEscapes: true,
				skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
			},
			messageStyle: "none",
			"HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
		});
		</script>
		<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	</head>
	<body>
		<div class="off-canvas-wrap">
			<div class="inner-wrap">
				<!-- mobile top bar to activate nav -->
				<nav class="tab-bar show-for-small">
					<section class="left-small">
						<a class="left-off-canvas-toggle menu-icon" ><span></span></a>
					</section>

					<section class="middle tab-bar-section">
						<h1 class="title">bonohu&nbsp;blog</h1>
					</section>
				</nav>

				<!-- mobile side bar nav -->
				<aside class="left-off-canvas-menu">
					<ul class="off-canvas-list">
						<li><a href="">Home</a></li>
						<li><label>Categories</label></li>
							<li ><a href="/category/dbcls.html">DBCLS</a></li>
							<li ><a href="/category/misc.html">misc</a></li>
							<li class="active"><a href="/category/rnaseq.html">rnaseq</a></li>
							<li ><a href="/category/shell.html">shell</a></li>

						<li><label>Links</label></li>
							<li><a href="http://bonohu.jp/">Dr.Bono's website</a></li>
							<li><a href="https://europepmc.org/authors/0000-0003-4413-0651">Europe PMC</a></li>
							<li><a href="https://github.com/bonohu">GitHub</a></li>
							<li><a href="http://orcid.org/0000-0003-4413-0651">ORCID</a></li>
							<li><a href="https://www.ncbi.nlm.nih.gov/pubmed?cmd=search&term=Bono%20H[au]%20AND%201995%3A2100[dp]%20NOT%20jpn[la]">PubMed</a></li>
							<li><a href="https://researchmap.jp/bonohu/?lang=english">ResearchMap</a></li>



						<li><label>Social</label></li>
							<li><a href="https://figshare.com/authors/Hidemasa_Bono/476712">Figshare</a></li>
							<li><a href="https://scholar.google.co.jp/citations?user=e6OaeXQAAAAJ&hl=en">GoogleScholar</a></li>
							<li><a href="https://profiles.impactstory.org/u/0000-0003-4413-0651">ImpactStory</a></li>
							<li><a href="https://www.researchgate.net/profile/Hidemasa_Bono">ResearchGate</a></li>
							<li><a href="https://twitter.com/drbonohu">Twitter</a></li>
					</ul>	
				</aside>

				<!-- top bar nav -->
				<nav class="top-bar hide-for-small-only" data-topbar>
					<ul class="title-area">
						<li class="name">
							<h1><a href="/">bonohu blog</a></h1>
						</li>
					</ul>

					<section class="top-bar-section">
						<ul class="left">
								<li ><a href="/category/dbcls.html">DBCLS</a></li>
								<li ><a href="/category/misc.html">misc</a></li>
								<li class="active"><a href="/category/rnaseq.html">rnaseq</a></li>
								<li ><a href="/category/shell.html">shell</a></li>
						</ul>
                        <ul class="right">                                                                                                                                           
                                                                                                                                             
                        </ul>  
					</section>
				</nav>

				<!-- Main Page Content and Sidebar -->
				<section class="main-section">
					<div class="row">
						<!-- Main Content -->
						<div class="medium-9 small-12 columns" role="content">
							<div class="row">
								<div class="small-12 columns">
									<h2>Category: rnaseq</h2>
								</div>
							</div>

<article>
	<h3><a href="/running-salmon-via-cwl.html">Running salmon via CWL</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 金 28 6月 2019.</h6>
	<p>If a transcriptome sequence set is available for the organism, <a href="https://combine-lab.github.io/salmon/">salmon</a> can be used for transcript quantification.
<code>cwltool</code> must be installed in advance by <code>conda install cwltool</code> or <code>pip install cwltool</code>.</p>
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="ch">#!/bin/sh</span>
<span class="c1"># number of …</span></pre></div></td></tr></table>
		<p class="continue"><a href="/running-salmon-via-cwl.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/running-ikra.html">Running ikra</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 火 21 5月 2019.</h6>
	<p>Tried to use <a href="https://github.com/yyoshiaki/ikra/">ikra</a>.
SRR mode which directly fetches SRA file from NCBI worked fine, but FASTQ mode
was failed after trimming with the error below.
It seems that the file setting for salmon was incorrect.</p>
<div class="highlight"><pre><span></span><span class="nv">Exception</span> : [
<span class="nv">The</span> <span class="nv">following</span> <span class="nv">errors</span> <span class="nv">were</span> <span class="nv">detected</span> <span class="nv">with</span> <span class="nv">the</span> <span class="nv">read</span> <span class="nv">files</span>
<span class="o">======================================================</span>
<span class="nv">ERROR</span>: <span class="nv">file</span> [<span class="nv">fq</span><span class="o">/</span><span class="nv">ERRxxxxxx_trimmed …</span></pre></div>
		<p class="continue"><a href="/running-ikra.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/fasterq-dump.html">fasterq-dump</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 土 02 2月 2019.</h6>
	<p>I noticed the announcement in <a href="https://github.com/ncbi/sra-tools/wiki">NCBI's sra-tools GitHub wiki</a></p>
<blockquote>
<p>With release 2.9.1 of sra-tools we have finally made available the tool <code>fasterq-dump</code>, a replacement for the much older <code>fastq-dump</code> tool. </p>
</blockquote>
<p>So I tested the speed from my home.</p>
<ol>
<li>Just specify a run ID of SRA.</li>
</ol>
<div class="highlight"><pre><span></span><span class="o">#</span> <span class="n">Just</span> <span class="n">fasterq</span><span class="o">-</span><span class="n">dump …</span></pre></div>
		<p class="continue"><a href="/fasterq-dump.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/cellfishing.html">CellFishing</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 火 06 11月 2018.</h6>
	<p>I tried to use <a href="https://doi.org/10.1101/374462">CellFishing</a> written in <a href="https://julialang.org">Julia</a>. 
It was my first time to use Julia, so I started from the installation of Julia with Homebrew.</p>
<div class="highlight"><pre><span></span># <span class="nv">Install</span> <span class="nv">Julia</span>
<span class="nv">brew</span> <span class="nv">cask</span> <span class="nv">install</span> <span class="nv">julia</span>
# <span class="nv">Install</span> <span class="nv">CellFishing</span>
<span class="nv">julia</span> <span class="o">-</span><span class="nv">e</span> <span class="s1">&#39;</span><span class="s">using Pkg; Pkg.add(PackageSpec(url=&quot;git://github.com/bicycle1885/CellFishing.jl.git&quot;))</span><span class="s1">&#39;</span>
# <span class="nv">Run …</span></pre></div>
		<p class="continue"><a href="/cellfishing.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/trinity-will-not-run.html">Trinity will not run</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 水 06 6月 2018.</h6>
	<p>After some software updates, Trinity ended with error below.</p>
<blockquote>
<p>dyld: Library not loaded: /usr/local/opt/gcc/lib/gcc/7/libgomp.1.dylib                                  <br>
Referenced from: /usr/local/Cellar/trinity/2.5.1/util/..//Inchworm/bin/fastaToKmerCoverageStats
Reason: image not found</p>
</blockquote>
<p>I checked the file <code>/usr/local/opt/gcc/lib/gcc …</code></p>
		<p class="continue"><a href="/trinity-will-not-run.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/stringtie.html">StringTie</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 金 18 5月 2018.</h6>
	<p>After the execution of HISAT2, aligned reads can be assembled using <a href="http://www.ccb.jhu.edu/software/stringtie/">StringTie</a>.</p>
<blockquote>
<p>StringTie is a fast and highly efficient assembler of RNA-Seq alignments into potential transcripts.</p>
</blockquote>
<p>The pipeline to process data by HISAT2 and StringTie is similar to that by TopHat and Cufflinks. Genome annotation file (GTF) can be used …</p>
		<p class="continue"><a href="/stringtie.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/hisat2.html">HISAT2</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 木 17 5月 2018.</h6>
	<p>HISAT2 is a successor of <a href="https://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a>.</p>
<blockquote>
<p>TopHat has entered a low maintenance, low support stage as it is now largely superseded by HISAT2 which provides the same core functionality (i.e. spliced alignment of RNA-Seq reads), in a more accurate and much more efficient way.</p>
</blockquote>
<p>Genome annotation file (GTF) is …</p>
		<p class="continue"><a href="/hisat2.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/align-and-estimate-abundance.html">Align and estimate abundance</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 水 07 3月 2018.</h6>
	<p>If assembled transcriptome sequence set and RNA-seq reads for that are available for an organism, we can align reads and estimate transcript abundance by running the script (<code>align_and_estimate_abundance.pl</code>) in the Trinity software package. When the Trinity package was installed using Homebrew, that script is installed in <code>/usr/local/Cellar …</code></p>
		<p class="continue"><a href="/align-and-estimate-abundance.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/running-kallisto.html">Running kallisto</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 月 22 1月 2018.</h6>
	<p>If a transcriptome sequence set is available for the organism, <a href="https://pachterlab.github.io/kallisto/">kallisto</a> can be used for transcript quantification. The <code>kallisto</code> version I used was <code>0.43.1</code>, which was previously installed using Homebrew by the command like <code>brew install kallisto</code>.</p>
<div class="highlight"><pre><span></span><span class="n">kallisto</span> <span class="k">index</span> <span class="o">-</span><span class="n">i</span> <span class="k">index</span> <span class="n">transcriptome</span><span class="p">.</span><span class="n">fa</span>
<span class="n">kallisto</span> <span class="n">quant</span> <span class="o">-</span><span class="n">i</span> <span class="k">index</span> <span class="o">-</span><span class="n">o …</span></pre></div>
		<p class="continue"><a href="/running-kallisto.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/installing-sleuth.html">Installing sleuth</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/rnaseq.html">rnaseq</a> on 月 22 1月 2018.</h6>
	<p>Following <a href="https://pachterlab.github.io/sleuth/download">instruction</a> in sleuth official page,
sleuth was successfully install except there was a permission problem in my <code>/usr/local</code>.<br>
I fixed that with a command, <code>sudo chown -R bono /usr/local</code>.</p>
		<p class="continue"><a href="/installing-sleuth.html">Continue reading &raquo;</a></p>
</article>
<hr />


						</div>
						<!-- End Main Content -->
						<!-- Sidebar -->
						<aside class="medium-3 hide-for-small-only columns">
							<div class="panel">
								<h5>Links</h5>
								<ul class="side-nav">
									<li><a href="http://bonohu.jp/">Dr.Bono's website</a></li>
									<li><a href="https://europepmc.org/authors/0000-0003-4413-0651">Europe PMC</a></li>
									<li><a href="https://github.com/bonohu">GitHub</a></li>
									<li><a href="http://orcid.org/0000-0003-4413-0651">ORCID</a></li>
									<li><a href="https://www.ncbi.nlm.nih.gov/pubmed?cmd=search&term=Bono%20H[au]%20AND%201995%3A2100[dp]%20NOT%20jpn[la]">PubMed</a></li>
									<li><a href="https://researchmap.jp/bonohu/?lang=english">ResearchMap</a></li>
								</ul>
							</div>

							<div class="panel">
								<h5>Tags</h5>
								<ul class="tag-cloud">
								</ul>
							</div>


							<div class="panel">
								<h5>Social</h5>
								<ul class="side-nav">
									<li><a href="https://figshare.com/authors/Hidemasa_Bono/476712">Figshare</a></li>
									<li><a href="https://scholar.google.co.jp/citations?user=e6OaeXQAAAAJ&hl=en">GoogleScholar</a></li>
									<li><a href="https://profiles.impactstory.org/u/0000-0003-4413-0651">ImpactStory</a></li>
									<li><a href="https://www.researchgate.net/profile/Hidemasa_Bono">ResearchGate</a></li>
									<li><a href="https://twitter.com/drbonohu">Twitter</a></li>
								</ul>
							</div>
						</aside>
						<!-- End Sidebar -->
					</div>

					<!-- Footer -->
					<footer class="row">
						<div class="medium-9 small-12">
							<hr/>
							<p class="text-center">Powered by <a href="http://getpelican.com">Pelican</a> and <a href="http://foundation.zurb.com/">Zurb Foundation</a>. Theme by <a href="http://hamaluik.com">Kenton Hamaluik</a>.</p>
						</div>
					</footer>
					<!-- End Footer -->
				</section>
				<a class="exit-off-canvas"></a>
			</div><!--off-canvas inner-->
		</div><!--off-canvas wrap-->

		<script src="/theme/js/jquery.js"></script>
		<script src="/theme/js/foundation.min.js"></script>
		<script>
			$(document).foundation();
		</script>
	</body>
</html>