<!doctype html>
<html class="no-js" lang="en">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />

		<title>bonohu blog</title>
		<meta name="description" content="">
		<meta name="author" content="bonohu">

		<link rel="stylesheet" href="/theme/css/foundation.css" />
		<link rel="stylesheet" href="/theme/css/pygment/monokai.css" />
		<link rel="stylesheet" href="/theme/css/custom.css" />


		<script src="/theme/js/modernizr.js"></script>

		<!-- Feeds -->

<script async src="https://www.googletagmanager.com/gtag/js?id=UA-113297-11"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-113297-11');
</script>

		<!-- mathjax config similar to math.stackexchange -->
		<script type="text/x-mathjax-config">
		MathJax.Hub.Config({
			jax: ["input/TeX", "output/HTML-CSS"],
			tex2jax: {
				inlineMath: [ ['$', '$'] ],
				displayMath: [ ['$$', '$$']],
				processEscapes: true,
				skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
			},
			messageStyle: "none",
			"HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
		});
		</script>
		<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	</head>
	<body>
		<div class="off-canvas-wrap">
			<div class="inner-wrap">
				<!-- mobile top bar to activate nav -->
				<nav class="tab-bar show-for-small">
					<section class="left-small">
						<a class="left-off-canvas-toggle menu-icon" ><span></span></a>
					</section>

					<section class="middle tab-bar-section">
						<h1 class="title">bonohu&nbsp;blog</h1>
					</section>
				</nav>

				<!-- mobile side bar nav -->
				<aside class="left-off-canvas-menu">
					<ul class="off-canvas-list">
						<li><a href="">Home</a></li>
						<li><label>Categories</label></li>
							<li ><a href="/category/dbcls.html">DBCLS</a></li>
							<li ><a href="/category/misc.html">misc</a></li>
							<li ><a href="/category/rnaseq.html">rnaseq</a></li>
							<li class="active"><a href="/category/shell.html">shell</a></li>

						<li><label>Links</label></li>
							<li><a href="http://bonohu.jp/">Dr.Bono's website</a></li>
							<li><a href="https://europepmc.org/authors/0000-0003-4413-0651">Europe PMC</a></li>
							<li><a href="https://github.com/bonohu">GitHub</a></li>
							<li><a href="http://orcid.org/0000-0003-4413-0651">ORCID</a></li>
							<li><a href="https://www.ncbi.nlm.nih.gov/pubmed?cmd=search&term=Bono%20H[au]%20AND%201995%3A2100[dp]%20NOT%20jpn[la]">PubMed</a></li>
							<li><a href="https://researchmap.jp/bonohu/?lang=english">ResearchMap</a></li>



						<li><label>Social</label></li>
							<li><a href="https://figshare.com/authors/Hidemasa_Bono/476712">Figshare</a></li>
							<li><a href="https://scholar.google.co.jp/citations?user=e6OaeXQAAAAJ&hl=en">GoogleScholar</a></li>
							<li><a href="https://profiles.impactstory.org/u/0000-0003-4413-0651">ImpactStory</a></li>
							<li><a href="https://www.researchgate.net/profile/Hidemasa_Bono">ResearchGate</a></li>
							<li><a href="https://twitter.com/drbonohu">Twitter</a></li>
					</ul>	
				</aside>

				<!-- top bar nav -->
				<nav class="top-bar hide-for-small-only" data-topbar>
					<ul class="title-area">
						<li class="name">
							<h1><a href="/">bonohu blog</a></h1>
						</li>
					</ul>

					<section class="top-bar-section">
						<ul class="left">
								<li ><a href="/category/dbcls.html">DBCLS</a></li>
								<li ><a href="/category/misc.html">misc</a></li>
								<li ><a href="/category/rnaseq.html">rnaseq</a></li>
								<li class="active"><a href="/category/shell.html">shell</a></li>
						</ul>
                        <ul class="right">                                                                                                                                           
                                                                                                                                             
                        </ul>  
					</section>
				</nav>

				<!-- Main Page Content and Sidebar -->
				<section class="main-section">
					<div class="row">
						<!-- Main Content -->
						<div class="medium-9 small-12 columns" role="content">
							<div class="row">
								<div class="small-12 columns">
									<h2>Category: shell</h2>
								</div>
							</div>

<article>
	<h3><a href="/tiny-perl-script-for-formatting-tsa-entries.html">Tiny Perl script for formatting TSA entries</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 月 03 9月 2018.</h6>
	<p>When we submit relatively large entries to DDBJ, we use <a href="https://www.ddbj.nig.ac.jp/ddbj/mss-e.html">Mass Submission System (MSS)</a>.
We will use MSS for submitting <a href="https://www.ddbj.nig.ac.jp/ddbj/tsa-e.html">transcriptome sequence assembly (TSA)</a> to DDBJ.</p>
<p>After extracting the list of IDs from header lines of FASTA by command like </p>
<p><code>% perl -nle 'print $1 if(/^\&gt;(\S+)/) hoge.fasta &gt; id.txt' …</code></p>
		<p class="continue"><a href="/tiny-perl-script-for-formatting-tsa-entries.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/removing-version-information-in-ids.html">Removing version information in IDs</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 金 31 8月 2018.</h6>
	<p>Identifiers (IDs) in public databases often contain version information.
For example, <code>.16</code> in <code>ENSG00000100644.16</code> from Ensembl and <code>.1</code> in <code>NM_001243084.1</code> from RefSeq.
Such version information can be an obstacle to join entries from different databases.
So, version information should be trimmed before joining.
The file that contains such …</p>
		<p class="continue"><a href="/removing-version-information-in-ids.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/local-reverse-complement.html">Local reverse complement</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 日 29 7月 2018.</h6>
	<p>To get reverse complement for specific DNA string is frequently needed in molecular biology.
There were some web interface to do that, but it is not secure.
Calculating reverse complement locally is ideal solution for that issue.</p>
<p>In the search of an example code with practical usefulness in GitHub,
I …</p>
		<p class="continue"><a href="/local-reverse-complement.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/moving-to-cleanly-installed-high-sierra.html">Moving to cleanly installed High Sierra</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 木 12 7月 2018.</h6>
	<p>As there was a mechanical trouble in my main machine, MacBookPro, I am moving to new MacBookPro with cleanly installed High Sierra (10.13.6). Below is a log for my future replication...</p>
<p>First of all, default shell was changed with <code>chsh</code> command to <code>/bin/zsh</code>.</p>
<p>After installing Homebrew, <code>coreutils …</code></p>
		<p class="continue"><a href="/moving-to-cleanly-installed-high-sierra.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/join-command-option-for-generating-tab-delimited-file.html">join command option for generating tab-delimited file</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 金 15 6月 2018.</h6>
	<p>When we want to join two files by a same key into one file, we can use <code>join</code> command. We can join lines by a first column value of tab-delimited files.</p>
<div class="highlight"><pre><span></span>join -j 1 file1 file2
</pre></div>


<p>Indeed, <code>join</code> command itself is very useful, default output is not tab-delimited text, but …</p>
		<p class="continue"><a href="/join-command-option-for-generating-tab-delimited-file.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/join-files-by-key.html">Join files by key</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 日 25 3月 2018.</h6>
	<p>Joining two files by key in the first column of files can be easily done by using UNIX command below.</p>
<div class="highlight"><pre><span></span>join -j 1 file1.txt file2.txt
</pre></div>


<p>This is very useful command, but the output is space-delimited by default.
In order to get the output by tab-delimited, following option for …</p>
		<p class="continue"><a href="/join-files-by-key.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/retrieve-a-subset-of-sequence-dataset.html">Retrieve a subset of sequence dataset</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 土 24 3月 2018.</h6>
	<p>In order to extract a set of sequence from FASTA-formatted file (both in nucleotides and peptides), several commands can be used to do so.
In recent years, I regularly use <code>blastdbcmd</code> in NCBI BLAST suite. To run this command, the file must be indexed by <code>makeblastdb</code> with the option below …</p>
		<p class="continue"><a href="/retrieve-a-subset-of-sequence-dataset.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/uniq-c-option.html">uniq -c option</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 金 23 3月 2018.</h6>
	<p>When I want to count the number of redundant words in a file (<code>hoge.txt</code>), I have used simple Perl code like this(<code>count.pl</code>).</p>
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="ch">#!/usr/bin/perl</span>
<span class="k">while</span><span class="p">(</span><span class="o">&lt;&gt;</span><span class="p">)</span> <span class="p">{</span>
        <span class="k">my</span><span class="p">(</span><span class="nv">$word</span><span class="p">)</span> <span class="o">=</span> <span class="nb">split</span><span class="p">;</span>
        <span class="nv">$num</span><span class="p">{</span><span class="nv">$word</span><span class="p">}</span><span class="o">++</span><span class="p">;</span>
<span class="p">}</span>
<span class="k">foreach</span> <span class="p">(</span><span class="nb">sort</span> <span class="nb">keys</span> <span class="nv">%num</span><span class="p">)</span> <span class="p">{</span>
        <span class="k">print</span> <span class="s">&quot;$_\t$num …</span></pre></div></td></tr></table>
		<p class="continue"><a href="/uniq-c-option.html">Continue reading &raquo;</a></p>
</article>
<hr />
<article>
	<h3><a href="/perl-in-awk-mode.html">Perl in awk mode</a></h3>
	<h6>Written by <a href="/author/bonohu.html">bonohu</a> in <a href="/category/shell.html">shell</a> on 月 05 2月 2018.</h6>
	<p>When I want to extract data by numeric value in other column, Perl in awk mode might be useful. It is often the case that I forget the option for that in Perl, so I took a note this time.</p>
<div class="highlight"><pre><span></span>perl -anle  &#39;print &quot;$F[0]\t$F[3]&quot; if ($F …</pre></div>
		<p class="continue"><a href="/perl-in-awk-mode.html">Continue reading &raquo;</a></p>
</article>
<hr />


						</div>
						<!-- End Main Content -->
						<!-- Sidebar -->
						<aside class="medium-3 hide-for-small-only columns">
							<div class="panel">
								<h5>Links</h5>
								<ul class="side-nav">
									<li><a href="http://bonohu.jp/">Dr.Bono's website</a></li>
									<li><a href="https://europepmc.org/authors/0000-0003-4413-0651">Europe PMC</a></li>
									<li><a href="https://github.com/bonohu">GitHub</a></li>
									<li><a href="http://orcid.org/0000-0003-4413-0651">ORCID</a></li>
									<li><a href="https://www.ncbi.nlm.nih.gov/pubmed?cmd=search&term=Bono%20H[au]%20AND%201995%3A2100[dp]%20NOT%20jpn[la]">PubMed</a></li>
									<li><a href="https://researchmap.jp/bonohu/?lang=english">ResearchMap</a></li>
								</ul>
							</div>

							<div class="panel">
								<h5>Tags</h5>
								<ul class="tag-cloud">
								</ul>
							</div>


							<div class="panel">
								<h5>Social</h5>
								<ul class="side-nav">
									<li><a href="https://figshare.com/authors/Hidemasa_Bono/476712">Figshare</a></li>
									<li><a href="https://scholar.google.co.jp/citations?user=e6OaeXQAAAAJ&hl=en">GoogleScholar</a></li>
									<li><a href="https://profiles.impactstory.org/u/0000-0003-4413-0651">ImpactStory</a></li>
									<li><a href="https://www.researchgate.net/profile/Hidemasa_Bono">ResearchGate</a></li>
									<li><a href="https://twitter.com/drbonohu">Twitter</a></li>
								</ul>
							</div>
						</aside>
						<!-- End Sidebar -->
					</div>

					<!-- Footer -->
					<footer class="row">
						<div class="medium-9 small-12">
							<hr/>
							<p class="text-center">Powered by <a href="http://getpelican.com">Pelican</a> and <a href="http://foundation.zurb.com/">Zurb Foundation</a>. Theme by <a href="http://hamaluik.com">Kenton Hamaluik</a>.</p>
						</div>
					</footer>
					<!-- End Footer -->
				</section>
				<a class="exit-off-canvas"></a>
			</div><!--off-canvas inner-->
		</div><!--off-canvas wrap-->

		<script src="/theme/js/jquery.js"></script>
		<script src="/theme/js/foundation.min.js"></script>
		<script>
			$(document).foundation();
		</script>
	</body>
</html>